doc/pages/architecture/090-database.md

```diff
@@ -150,52 +159,101 @@ CREATE TABLE <realm name>.interfaces (
   aggregation int,
   automaton_transitions blob,
   automaton_accepting_states blob,
-  description varchar,
-  doc varchar,
+  description text,
+  doc text,
 
   PRIMARY KEY (name, major_version)
 );
 ```
 
+```sql
+CREATE TABLE <realm name>.simple_triggers (
+  object_id uuid,
+  object_type int,
+  parent_trigger_id uuid,
+  simple_trigger_id uuid,
+  trigger_data blob,
+  trigger_target blob,
+
+  PRIMARY KEY ((object_id, object_type), parent_trigger_id, simple_trigger_id)
+);
+```
+
+```sql
+
+CREATE TABLE <realm name>.individual_datastreams (
+    device_id uuid,
+    interface_id uuid,
+    endpoint_id uuid,
+    path text,
+    value_timestamp timestamp,
+    reception_timestamp timestamp,
+    reception_timestamp_submillis smallint,
+    binaryblob_value blob,
+    binaryblobarray_value list<blob>,
+    boolean_value boolean,
+    booleanarray_value list<boolean>,
+    datetime_value timestamp,
+    datetimearray_value list<timestamp>,
+    double_value double,
+    doublearray_value list<double>,
+    integer_value int,
+    integerarray_value list<int>,
+    longinteger_value bigint,
+    longintegerarray_value list<bigint>,
+    string_value text,
+    stringarray_value list<text>,
+    PRIMARY KEY ((device_id, interface_id, endpoint_id, path), value_timestamp, reception_timestamp, reception_timestamp_submillis)
+) 
+```
+
 ```sql
 CREATE TABLE <realm name>.individual_properties (
   device_id uuid,
   interface_id uuid,
   endpoint_id uuid,
-  path varchar,
+  path text,
   reception_timestamp timestamp,
   reception_timestamp_submillis smallint,
-
   double_value double,
   integer_value int,
   boolean_value boolean,
   longinteger_value bigint,
-  string_value varchar,
+  string_value text,
   binaryblob_value blob,
   datetime_value timestamp,
   doublearray_value list<double>,
   integerarray_value list<int>,
   booleanarray_value list<boolean>,
   longintegerarray_value list<bigint>,
-  stringarray_value list<varchar>,
+  stringarray_value list<text>,
   binaryblobarray_value list<blob>,
   datetimearray_value list<timestamp>,
 
   PRIMARY KEY((device_id, interface_id), endpoint_id, path)
 );
+
 ```
+Following templates are generated upon datastream interface creation for keeping all data sent to astarte through the interface.
 
+Table name comes from lower case interface name where ```.``` and ```-``` have been replaced by ```_``` and ```""``` (empty string), then interface's major version is appended with a ```_v``` prefix. 
+For example, com.Astarte.TestInterface version 1 becomes  ```com_astarte_testinterface_v1```.
+
+If, after all the requiret transformations, the resulting name is too long (>45 chars), it's going to be encoded and truncated
```

Comment:
requiret --> required

"it will be encoded and truncated" sounds better than "it's going to...", imho